ynr/apps/sopn_parsing/helpers/extract_pages.py

```diff
@@ -22,8 +23,42 @@ def extract_pages_for_ballot(ballot, manual_upload=False):
     """
     if ballot.sopn.uploaded_file.path.endswith(".pdf"):
         pass
-    elif ballot.sopn.uploaded_file.path.endswith(".docx"):
-        # import pdb; pdb.set_trace()
+    elif ballot.sopn.uploaded_file.path.endswith(".html"):
+        print(
+            f"Warning! HTML file types are not supported; trying to convert {ballot.sopn.uploaded_file.path} to pdf"
+        )
+        html_file_path = ballot.sopn.uploaded_file.path
+        pdf_file_path = ballot.sopn.uploaded_file.path + ".pdf"
+        # Replacing the extention with .pdf also works and
+        # negates the need for pdfkit but not sure
+        # how reliable it is
+        # pdf_file = ballot.sopn.uploaded_file.path.replace(".html", ".pdf")
+        config = pdfkit.configuration(wkhtmltopdf="/usr/local/bin/wkhtmltopdf")
+        options = {"enable-local-file-access": None}
+
+        pdfkit.from_file(
+            html_file_path,
+            pdf_file_path,
+            configuration=config,
+            verbose=True,
+            options=options,
+        )
+
+        # save the pdf file as the sopn document uploaded file
+        # ballot.sopn.uploaded_file.save(
+        #     ballot.sopn.uploaded_file.name,
+        #     File(open(pdf_file_path, "rb")),
+        #     save=True,
+        # )
+        # ballot.sopn.save()
+
```

Comment:
File conversion now works, but I need to refactor this block that attempts to save the new pdf file path to the SOPN.